pkg/container/bubblewrap_runner.go

```diff
@@ -85,9 +85,16 @@ func (bw *bubblewrap) cmd(ctx context.Context, cfg *Config, debug bool, envOverr
 		"--chdir", runnerWorkdir,
 		"--clearenv")
 
+	// Set caps as for the current thread bounding capability set.
+	baseargs = append(baseargs, "--cap-add", "all")
+
```

Comment:
I will always advocate for least privileges.

If most things build without these privileges, can we remove them from docker and qemu runners too?

Reply:
I think it's doable. We'd need to adjust some pipeline, especially the melange stage 2 that tries to run mount. Cases where dac_override is needed I think we'd just need to set permissions properly on files, to avoid it.

Reply:
nice, but likely lots of small issues to open and tackle one at a time.

Reply:
@maxgio92 @xnox 
the problem here is that *before* the caps where *not* dropped (hence the boostrap was working)

Now that we fixed this, we should add the caps needed
What I would think here is that we use `--unshare-user` (this will drop caps) *only* is the original user is *not* root, this way we bypass the `sudo melange` not working

And we keep the drop + basic caps from here:  https://github.com/moby/moby/blob/master/oci/caps/defaults.go#L6-L19

Reply:
@89luca89 I wouldn't add by default any capability. We can end up without any issue.

In order to remove the inconsistency between runs without sudo, I think we should also, when not specified by RunAs, keep a standard uid by default instead of mapping the original user id, whether is 1000 (build) or 0 with no caps.

Reply:
There are some minimal caps that we need, mainly for stuff like chmod/own and basic shell, remember that we have literal shell stuff to run

I'd say that minimal subset should work, it's a sensible default
If we want more security we should use other stuff not bwrap in the first place

Reply:
After some investigation, I see that if not specificed (cfg.RunAs) we default to NO CAPS, with the regular user.

So we need to fix the user mapping only if cfg.RunAs AND if we're not using `sudo melange` for some reasons.

So let's skip the CAPs, and just work around this problem:

```go
	// if we need to run as an user, we run as that user
	if cfg.RunAs != "" {
		baseargs = append(baseargs, "--unshare-user")
		baseargs = append(baseargs, "--uid", cfg.RunAs)
		baseargs = append(baseargs, "--gid", cfg.RunAs)
	// else if we're not using melange as root, we force the use of the
	// apko build user, this avoids problems on machines where default
	// regular user is NOT 1000
	} else if os.Getuid() > 0 {
		baseargs = append(baseargs, "--unshare-user")
		baseargs = append(baseargs, "--uid", "1000")
		baseargs = append(baseargs, "--gid", "1000")
	}
```

This should fix your inconsistent behaviour on non-1000 default users, but also don't change the default behaviour for present pipelines

Reply:
Agree, so we split the sudo vs unprivileged running user inconsistency scenario from the inconsistency due to the current uid vs standard uid (i.e. 1000).